src/host/node_connections.h

```diff
@@ -94,56 +111,58 @@ namespace asynchost
       size_t id;
 
       IncomingBehaviour(NodeConnections& parent, size_t id) :
-        ConnectionBehaviour(parent, ccf::NoNode),
+        ConnectionBehaviour(parent),
         id(id)
       {}
 
       void on_disconnect()
       {
-        LOG_DEBUG_FMT("node incoming disconnect {} with node {}", id, node);
-
         parent.incoming.erase(id);
 
-        if (node != ccf::NoNode)
-          parent.associated.erase(node);
+        if (node.has_value())
+        {
+          LOG_DEBUG_FMT(
+            "node incoming disconnect {} with node {}", id, node.value());
+          parent.associated.erase(node.value());
+        }
       }
 
-      virtual void associate(ccf::NodeId n)
+      virtual void associate(const ccf::NodeId& n)
       {
         node = n;
-        parent.associated.emplace(node, parent.incoming.at(id));
-        LOG_DEBUG_FMT("node incoming {} associated with {}", id, node);
+        parent.associated.emplace(node.value(), parent.incoming.at(id));
+        LOG_DEBUG_FMT("node incoming {} associated with {}", id, node.value());
       }
     };
 
     class OutgoingBehaviour : public ConnectionBehaviour
     {
     public:
-      OutgoingBehaviour(NodeConnections& parent, ccf::NodeId node) :
+      OutgoingBehaviour(NodeConnections& parent, const ccf::NodeId& node) :
         ConnectionBehaviour(parent, node)
       {}
 
       void on_resolve_failed()
       {
-        LOG_DEBUG_FMT("node resolve failed {}", node);
+        LOG_DEBUG_FMT("node resolve failed {}", node.value());
         reconnect();
       }
 
       void on_connect_failed()
       {
-        LOG_DEBUG_FMT("node connect failed {}", node);
+        LOG_DEBUG_FMT("node connect failed {}", node.value());
         reconnect();
       }
 
       void on_disconnect()
       {
-        LOG_DEBUG_FMT("node disconnect failed {}", node);
+        LOG_DEBUG_FMT("node disconnect failed {}", node.value());
```

Comment:
I think this should be `node` rather than `node.value()`, to use the custom formatter which prints only the first 10 characters?

Reply:
`node` here is actually a `std::optional` (since incoming channels are unassociated when created), so this will print the short hash.